core/core.go

```diff
@@ -87,8 +93,10 @@ type NetworkDriver interface {
 	DeleteNetwork(id string) error
 	CreateEndpoint(id string) error
 	DeleteEndpoint(id string) error
-	CreatePeerHost(id string) error
-	DeletePeerHost(id string) error
+	AddPeerHost(node ServiceInfo) error
```

Comment:
oh yeah I just meant embed it directly as follows but the current form is fine as well:

```
type NetworkDriver interface {
...
Clustering
}
```